db/migrate/20170728140134_remove_email_from_enterprises.rb

```diff
@@ -0,0 +1,48 @@
+class RemoveEmailFromEnterprises < ActiveRecord::Migration
+  class Enterprise < ActiveRecord::Base; end
+  class Spree::User < ActiveRecord::Base; end
+
+  class EnterpriseRole < ActiveRecord::Base
+    belongs_to :user, class_name: 'Spree::User'
+    belongs_to :enterprise, class_name: 'Enterprise'
+  end
+
+  def up
+    Enterprise.reset_column_information
+    Spree::User.reset_column_information
+    EnterpriseRole.reset_column_information
+
+    Enterprise.select([:id, :email, :owner_id]).each do |enterprise|
+      contact_user = Spree::User.find_by_email enterprise.email
+      unless contact_user
+        password = Devise.friendly_token.first(8)
+        contact_user = Spree::User.create(email: enterprise.email, password: password, password_confirmation: password)
+        contact_user.send_reset_password_instructions if contact_user.persisted?
+      end
+
+      unless contact_user.persisted?
+        contact_user = Spree::User.find enterprise.owner_id
+      end
+
+      manager = EnterpriseRole.find_or_initialize_by_user_id_and_enterprise_id(contact_user.id, enterprise.id)
+      manager.update_attribute :receives_notifications, true
+    end
+
+    remove_columns :enterprises, :email, :contact
```

Comment:
This `contact` here is supposed to be just a profile email address, not connected to any user.

Reply:
Maybe we should add this info in some comment in the code

Reply:
There's another field which I think is called `email_address` which has not been removed, and is for displaying in the profile as you mentioned.

Reply:
Previously `email` was an email address and `contact` was a string for the name of the person the emails would be sent to. In addition to these, there is a separate field called `email_address`.

Reply:
@mkllnk  It's under the Contact tab in the Enterprise edit page.

Reply:
To clarify: `contact` is a string representing the *name* of the contact person for the enterprise. We need it, even if we need to rename it as `contact_name` to avoid a conflict with the new `contact` association.

Reply:
Renamed to `contact_name`.